tasks.c

```diff
@@ -889,13 +893,13 @@ static void prvYieldForTask( TCB_t * pxTCB,
 
                     if( pxTCB->xTaskRunState == taskTASK_NOT_RUNNING )
                     {
-                        #if ( configUSE_CORE_EXCLUSION == 1 )
-                            if( ( pxTCB->uxCoreExclude & ( 1 << xCoreID ) ) == 0 )
+                        #if ( configUSE_CORE_AFFINITY == 1 )
+                            if( ( pxTCB->uxCoreAffinityMask & ( 1 << xCoreID ) ) != 0 )
```

Comment:
Suggestion: Separate #if / #else / #endif sections would make this logic more clear.